apps/builder/app/shared/copy-paste/plugin-webflow/plugin-webflow.ts

```diff
@@ -110,31 +114,56 @@ export const onPaste = async (clipboardData: string) => {
     return false;
   }
   const fragment = await toWebstudioFragment(wfData);
+
   const selectedPage = $selectedPage.get();
-  if (fragment.instances.length === 0 || selectedPage === undefined) {
+  if (fragment === undefined || selectedPage === undefined) {
     return false;
   }
   const metas = $registeredComponentMetas.get();
   const newInstances = new Map(
     fragment.instances.map((instance) => [instance.id, instance])
   );
-  const rootInstanceIds = fragment.children
-    .filter((child) => child.type === "id")
-    .map((child) => child.value);
+
   // paste to the root if nothing is selected
   const instanceSelector = $selectedInstanceSelector.get() ?? [
     selectedPage.rootInstanceId,
   ];
+
+  const rootInstanceIds = fragment.children
+    .filter((child) => child.type === "id")
+    .map((child) => child.value);
+
   const dropTarget = findClosestDroppableTarget(
     metas,
     $instances.get(),
     instanceSelector,
     computeInstancesConstraints(metas, newInstances, rootInstanceIds)
   );
+
   if (dropTarget === undefined) {
     return false;
   }
-  insertTemplateData(fragment, dropTarget);
+
+  updateWebstudioData((data) => {
+    const { newInstanceIds } = insertWebstudioFragmentCopy({
+      data,
+      fragment,
+      availableDataSources: findAvailableDataSources(
+        data.dataSources,
+        data.instances,
+        instanceSelector
+      ),
+    });
+    const newRootInstanceId = newInstanceIds.get(fragment.instances[0].id);
+    if (newRootInstanceId === undefined) {
+      return;
+    }
+    const children: Instance["children"] = [
+      { type: "id", value: newRootInstanceId },
+    ];
+    insertInstanceChildrenMutable(data, children, dropTarget);
```

Comment:
@TrySound need your help here, newInstanceIds doesn't return the right items, also my code on ln 157 is wrong too, I guess it needs something like this:

```js
const children = fragment.children
      .map((child) => {
        if (child.type !== "id") {
          return;
        }
        return {
          type: "id",
          value: newInstanceIds.get(child.value),
        };
      })
      .filter(Boolean);

```